fre/pp/split_netcdf_script.py

```diff
@@ -0,0 +1,289 @@
+  component - the 'component' element we are currently working with in the yaml
+  history_source - a history_file under a 'source' under the 'component' that
+    we are working with. Is used to identify the files in inputDir.
+  use_subdirs - whether to recursively search through inputDir under the subdirectories.
+    used when regridding.
+  yamlfile - a .yml config file for fre postprocessing
+  '''
+  
+  #Verify input/output dirs exist and are dirs
+  if not (os.path.isdir(inputDir)):
+    fre_logger.error(f"error: input dir {inputDir} does not exist or is not a directory")
+    raise OSError(f"error: input dir {inputDir} does not exist or is not a directory")
+  if not (os.path.isdir(outputDir)):
+    fre_logger.error(f"error: output dir {outputDir} does not exist or is not a directory")
+    raise OSError(f"error: output dir {outputDir} does not exist or is not a directory")
+  
+  #Find files to split
+  curr_dir = os.getcwd()
+  workdir = os.path.abspath(inputDir)
+  
+  #note to self: if CYLC_TASK_PARAM_component isn't doing what we think it's
+  #doing, we can also use history_source to get the component but it's
+  #going to be a bit of a pain
+  varlist = parse_yaml_for_varlist(yamlfile, component, history_source)
+  
+  #extend globbing used to find both tiled and non-tiled files
+  #all files that contain the current source:history_file name,
+  #0-1 instances of "tile" and end in .nc
+  #under most circumstances, this should match 1 file
+  #older regex - not currently working
+  #file_regex = f'*.{history_source}?(.tile?).nc'
+  file_regex = f'*.{history_source}*.*.nc'
+  
+  #If in sub-dir mode, process the sub-directories instead of the main one
+  # and write to $outputdir/$subdir
+  if use_subdirs:
+    subdirs = [el for el in os.listdir(workdir) if os.path.isdir(el)]
+    num_subdirs = len(subdirs)
+    fre_logger.info(f"checking {num_subdirs} under {workdir}")
+    files_split = 0
+    for sd in subdirs: 
+      files=glob.glob(os.path.join(sd,file_regex))
+      if len(files) == 0:
+        fre_logger.info(f"No input files found; skipping subdir {subdir}")
+      else:
+        output_subdir = os.path.join(os.path.abspath(outputDir)/sd)
+        if not os.path.isdir(ouput_subdir):
+          os.mkdir(output_subdir)
+        for infile in files:
+          split_file_xarray(infile, output_subdir, varlist)
+          files_split += 1
+    fre_logger.info(f"{files_split} files split")
+    if files_split == 0:
+      fre_logger.warning(f"warning: no files found in dirs under {workdir} that match pattern {file_regex}; no splitting took place")
+  else:
+      files=glob.glob(os.path.join(workdir, file_regex))
+      # Split the files by variable
+      for infile in files:
+        split_file_xarray(infile, os.path.abspath(outputDir), varlist)
+      if len(files) == 0:
+        fre_logger.warning(f"warning: no files found in {workdir} that match pattern {file_regex}; no splitting took place")
```

Comment:
Are there expected scenarios where we call the splitter and it ends up doing nothing? If the answer is definitely no, then we should raise an exception here instead of a warning.